### PR TITLE
SNOW-1269037: [Local Testing] Add support for NaT and NaN values

### PR DESCRIPTION
**Pull request by sfc-gh-jrose**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   SNOW-1269037

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   This PR adds support for the None like values NaN (not a number) and NaT (not a time) to the schema inference function for local testing.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +40,10 @@
 ### Bug Fixes
 
 - Fixed a bug where `statement_params` was not passed to query executions that register stored procedures and user defined functions.
+- Fixed a bug causing `snowflake.snowpark.Session.file.get_stream` to fail for quoted stage locations
```
Comment (sfc-gh-stan) [inline CHANGELOG.md:43]: These entries seem unrelated.

File: src/snowflake/snowpark/_internal/type_utils.py [hunk 2]
```diff
@@ -264,6 +264,15 @@ def convert_sp_to_sf_type(datatype: DataType) -> str:
     datetime.time: TimeType,
     bytes: BinaryType,
 }
+if installed_pandas:
```
Comment (sfc-gh-stan) [inline src/snowflake/snowpark/_internal/type_utils.py:267]: Elegant :chef_kiss:

**Review by sfc-gh-stan — APPROVED** [review]
Code LGTM, please update the changelog.

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,10 @@
 
 - Added support for StringType, TimestampType and VariantType data conversion in the mocked function `to_date`.
 
+### Bug Fixes
```
Comment (sfc-gh-aling) [inline CHANGELOG.md:15]: nit:
```suggestion
#### Bug Fixes
```

File: src/snowflake/snowpark/_internal/type_utils.py [hunk 4]
```diff
@@ -264,6 +264,15 @@ def convert_sp_to_sf_type(datatype: DataType) -> str:
...
+    import numpy
+
+    PYTHON_TO_SNOW_TYPE_MAPPINGS.update(
+        {
+            pandas._libs.tslibs.nattype.NaTType: TimestampType,
```
Comment (sfc-gh-aling) [inline src/snowflake/snowpark/_internal/type_utils.py:272]: is there a public api form pandas for the `NaTType` or alternatives ways to do this depending on public pandas api?

I raise the question because I recently handled this issue: https://github.com/snowflakedb/snowpark-python/issues/1388.

Comment (sfc-gh-jrose) [reply src/snowflake/snowpark/_internal/type_utils.py:272]: I could make this change, but it seems a little less clean:
```suggestion
            type(pandas.NaT): TimestampType,
```

The main issue is that pandas recommended way of checking for nat is with pandas.isna or pandas.isnull which doesn't fit well with our generic approach here.

Comment (sfc-gh-aling) [reply src/snowflake/snowpark/_internal/type_utils.py:272]: I personal like this `type(pandas.NaT)` approach, let's make the change?

**Review by sfc-gh-aling — APPROVED** [review]
